app/src/main/java/mozilla/lockbox/store/ItemDetailStore.kt

```diff
@@ -154,13 +154,14 @@ open class ItemDetailStore(
     }
 
     private fun calculateUnavailableUsernames(getItem: Observable<Optional<ServerPassword>>, excludeItem: Boolean) {
-        Observables.combineLatest(getItem, dataStore.list)
+        Observables.combineLatest(getItem, dataStore.list.take(1))
             .map { (opt, list) ->
                 val item = opt.value ?: return@map emptySet<String>()
 
                 val usernames = list
                     .filter(
-                        hostname = item.hostname,
+                        // Default to a blank space, otherwise we get all usernames for any hostname.
+                        hostname = if (item.hostname.isNullOrBlank()) " " else item.hostname,
```

Comment:
Why would the `hostname` ever be null/blank? This doesn't seem like a valid case to me

Reply:
`hostname` is blank in the case of Manual Create.